tests/components/alarmdecoder/test_config_flow.py

```diff
@@ -0,0 +1,424 @@
+    assert entry.options == {
+        OPTIONS_ARM: user_input,
+        OPTIONS_ZONES: DEFAULT_ZONE_OPTIONS,
+    }
+
+
+async def test_options_zone_flow(hass: HomeAssistant):
+    """Test options flow for adding/deleting zones."""
+    zone_number = "2"
+    zone_settings = {CONF_ZONE_NAME: "Front Entry", CONF_ZONE_TYPE: DEVICE_CLASS_WINDOW}
+    entry = MockConfigEntry(domain=DOMAIN)
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "init"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={"edit_selection": "Zones"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_select"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={CONF_ZONE_NUMBER: zone_number},
+    )
+
+    with patch(
+        "homeassistant.components.alarmdecoder.async_setup_entry", return_value=True
+    ):
+        result = await hass.config_entries.options.async_configure(
+            result["flow_id"],
+            user_input=zone_settings,
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert entry.options == {
+        OPTIONS_ARM: DEFAULT_ARM_OPTIONS,
+        OPTIONS_ZONES: {zone_number: zone_settings},
+    }
+
+    # Make sure zone can be removed...
+    result = await hass.config_entries.options.async_init(entry.entry_id)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "init"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={"edit_selection": "Zones"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_select"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={CONF_ZONE_NUMBER: zone_number},
+    )
+
+    with patch(
+        "homeassistant.components.alarmdecoder.async_setup_entry", return_value=True
+    ):
+        result = await hass.config_entries.options.async_configure(
+            result["flow_id"],
+            user_input={},
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert entry.options == {
+        OPTIONS_ARM: DEFAULT_ARM_OPTIONS,
+        OPTIONS_ZONES: {},
+    }
+
+
+async def test_options_zone_flow_validation(hass: HomeAssistant):
+    """Test input validation for zone options flow."""
+    zone_number = "2"
+    zone_settings = {CONF_ZONE_NAME: "Front Entry", CONF_ZONE_TYPE: DEVICE_CLASS_WINDOW}
+    entry = MockConfigEntry(domain=DOMAIN)
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
```

Comment:
We should set up the config entry before starting the options flow, to make sure the options flow is loaded. Otherwise we depend on the other tests to do that.